components/context_id/src/callback.rs

```diff
@@ -0,0 +1,15 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#[uniffi::export(callback_interface)]
+pub trait ContextIdCallback: Sync + Send {
+    fn persist(&self, context_id: String, creation_date: i64);
+    fn rotated(&self, old_context_id: String);
```

Comment:
I think these methods should return a `Result` and the `E` side should implement `From<uniffi::UnexpectedUniFFICallbackError>`.  Otherwise, if the foreign side throws for some unexpected reason the Rust code will panic.  The bottom part of https://mozilla.github.io/uniffi-rs/latest/foreign_traits.html has more details.

Reply:
I've gone ahead and tried to log errors when these callbacks fail. Does that seem like the right move?